### PR TITLE
SkiaSharp - Search for Fonts in EmbeddedResources

### LINKED ISSUE #1735: Unable to use custom font with SkiaSharp.Wpf
### Feature description

I should be able to use custom fonts from a local .ttf file using SkiaSharp.Wpf. 

Implementation suggestion:

In the current implementation of SkiaRenderContext, the DrawText method (and GetTextPaint method internally) use an SKTypeface cache (called typefaceCache). A method to add custom fonts to this cache would make custom fonts accessible. The SKTypeface.FromStream Method could be used in this process.

### PR DESCRIPTION
**Pull request by JensKrumsieck**
Adds EmbeddedResource based fonts to SkiaSharp Renderer as `SKTypeface.FromFamilyName` is not valid in e.g. WASM. 
Uses inspiration from https://www.blazorspread.net/blogview/82.
Tested with Blazor WASM but probably also Fixes #1735  .
Axis rendered with Montserrat Font in WASM
montserrat

### Checklist

- [ ] I have included examples or tests
- [ ] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file
- [ ] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Added Dictionary for translating fontweights (int) to string as this is used for naming .ttf file when downloaded from e.g. GoogleFonts 
  ```csharp
  private readonly Dictionary<int, string> _fontWeights = new Dictionary<int, string>()
          {
              [100] = "Thin",
              [200] = "ExtraLight",
              [300] = "Light",
              [400] = "Regular",
              [500] = "Medium",
              [600] = "SemiBold",
              [700] = "Bold",
              [800] = "ExtraBold",
              [900] = "Black"
          };
  ```
- Search embedded resources if font ist not found by name
  ```csharp
  if (typeface.FamilyName != fontFamily) // requested font not found or is WASM
                  {
                      try
                      {
                          var assembly = Assembly.GetEntryAssembly(); // the executing program (the GUI Project (WPF, WASM, ...))
                          var weight = (_fontWeights.ContainsKey((int)fontWeight) ? _fontWeights[(int)fontWeight] : "Regular");
                          var filename = $"{fontFamily}-{weight}.ttf".ToLower();
                          Console.WriteLine($"Load Font {filename}");
  
                          var matches = assembly!.GetManifestResourceNames().Where(item => item.ToLower().EndsWith(filename));
                          if (!matches.Any()) matches = assembly!.GetManifestResourceNames().Where(item => item.ToLower().EndsWith(fontFamily + ".ttf"));
                          foreach (var item in matches)
                          {
                              var s = assembly.GetManifestResourceStream(item);
                              typeface = SKTypeface.FromStream(s);
                          }
                      }
                      catch
                      {
                          Console.WriteLine($"Requested Font {fontFamily} could not be found, falling back to {typeface.FamilyName}");
                      }
                  }
  ```
@oxyplot/admins

### CODE REVIEW

**Comment by JensKrumsieck** [comment]
Regarding the CI Build error:

There is no `Assembly.GetEntryAssembly()`  in netstandard1.3, which is one of the TargetFrameworks of the SkiaSharp-Package. 
https://github.com/oxyplot/oxyplot/blob/e88c97fc1bc2c6da76ae45e2afd8531ee262ba89/Source/OxyPlot.SkiaSharp/SkiaRenderContext.cs#L873
The idea behind this line was to pack the Fonts into the UI Assemblies. 
To get rid of the Build error i see two possibilities:
1. Drop netstandard1.3 or increase to netstandard1.5 as `Assembly.GetEntryAssembly()` was added in the latter
2. Exclude this feature for netstandard1.3 with `#if NETSTANDARD1_5_OR_GREATER || NET452_OR_GREATER`

**Comment by VisualMelon** [comment]
@JensKrumsieck thanks for that discussion; I'll try to take a proper look at this over the weekend

**Comment by VisualMelon** [comment]
Sorry for taking so long to get back to this; I think excluding it for not-netstandard2.0 is the right thing to do (i.e. option 2).

If you're happy to make that change, then I'll approve this, and if none of the other maintainers object I'll merge a few days thereafter.

**Review by VisualMelon — APPROVED** [review]

**Comment by VisualMelon** [comment]
Sorry for taking so long to get back to this.

**Pull request merged**
